releases-implementation/README.md

```diff
@@ -0,0 +1,415 @@
+- after cluster-$provider app is applied successfully, in order to render all templates and set the app version, catalog and depends on properties, Helm will do the following:
+	- it will lookup the App resource (via Helm `lookup` function) and read release version,
+	- then it will lookup the Release resource and read app version, catalog and depends on properties from there, and
+	- finally it will render all templates and apply them.
+
+From this point forward, everything is the same like when all app details were set directly in the Helm chart.
+
+### 4.2. Comparison with the current releases
+
+This section compares the proposed release process with the current one, where we deploy clusters with cluster-$providers apps which have information about app versions.
+
+We cover some of the topics mentioned in section 3. Motivation.
+
+#### 4.2.1. Release identifier and versioning
+
+With the current cluster-$provider apps releases process, the cluster-$providers app version is the release identifier.
+
+With the releases repository, release identifiers would be defined in the repository itself, in the same way like for vintage releases. In this process the cluster-$provider app is one of the release components.
+
+In both cases there is a single release identifier, just defined in different places.
+
+Versioning logic, i.e. what and how can change in a major, minor  or a patch version, would be the same in both cases.
+
+#### 4.2.2. Creating and maintaining multiple releases
+
+Today, with the current cluster-$provider apps releases process, we only support the latest version of the cluster-$provider app. However this will change now, as we will have to support multiple major releases (e.g. to support multiple Kubernetes versions).
+
+As described in section *[3.2. Creating and maintaining multiple releases](#32-creating-and-maintaining-multiple-releases)*, we will be in a situation where we are supporting 10 or more major releases across all providers (5-6 providers, 2-3 major release per provider). In order to maintain all those major releases:
+- We need one cluster-$provider app git branch per major release, so for 10 major releases this means 10 git branches (across multiple repos).
+- We also need one git branch for every major version of the cluster chart.
+- In case we want to support more than 1 minor version for every  major, the number of git branches is even larger.
+
+##### 4.2.2.2. Scenario 1: new provider-independent app patch
+
+Now in the case above, let’s say we want to release a new patch version for some provider-independent app, and we have to do that in 5 providers, and for 2 major releases for every provider. The process would look like the following:
+- Renovate updates the version of the app in 2 git branches in the cluster chart. Here we have 2 pull requests, both created by Renovate.
```

Comment:
For reference as I wasn't aware - you can specify multiple base branches in Renovate, that's very nice! 😁  https://docs.renovatebot.com/configuration-options/#basebranches

Reply:
Yeah, if we keep the current way, then we would really have to do that. You still have PRs though.

And Renovate can't help if you have to cherry-pick entire changes/fixes/features to older release branches, and when you have to deal with git conflicts.